### ISSUE TITLE
Two requests take 7+ min (for world, 30s for Switzerland) and make whole site unusable

### ISSUE THREAD

**Issue by corneliusroemer** [GenSpectrum/cov-spectrum-website#333]
I know you like having all data available in the browser @chaoran-chen . But I do think this is really not great for users:
a) crazy amount of data transferred
b) slow like hell

All of this for one or two panels that are not that important, like hospitalization etc. Could we maybe make them optional? So that the large requests are only made when one _really_ wants that? It feels like we're carrying around a lot of neat but relatively unimportant things that weigh down performance of the really important bits.

![image](https://user-images.githubusercontent.com/25161793/140555184-fc9c40f3-cc57-41ec-b9fe-e09a770dd77d.png)

Request URL: https://cov-spectrum.ethz.ch/gisaid/api/v1/sample/aggregated?fields=date%2Cregion%2Ccountry%2Cdivision%2Cage%2Csex%2Chospitalized%2Cdied%2CfullyVaccinated&host=Human&dateFrom=2020-01-0

**Comment by corneliusroemer** [comment]
It keeps going... 7min now
![image](https://user-images.githubusercontent.com/25161793/140555474-d3b397ad-258d-4da7-b674-e8b30b50489f.png)

**Comment by chaoran-chen** [comment]
Is the delay caused by transfer or calculation? (Reducing the data transfer will likely increase the overall server calculation time because we would need to make multiple parallel requests instead of a single one)

I would first do the improvements in #298 before trying other stuff (because I don't think that having much data in the browser actually is the cause of the performance problems..).

**Comment by corneliusroemer** [comment]
I don't know, but I can't see how we need 400 MB... This is very unorthodox and not surprising it takes long.

What do we need all this data for?

Is it because it's

- a) daily data -> could switch to weekly with no real granularity hit
- b) metadata like hospitalization/vax status that slows it down? if so, chuck them out
- c) I don't think it's the receiving that's the problem, more than the serving?

Basically, for a few line plots, I don't understand why we need 500 MB...

**Comment by corneliusroemer** [comment]
The problem is this request which seems to deliver a lot of data that I really don't care about:
https://cov-spectrum.ethz.ch/gisaid/api/v1/sample/aggregated?fields=date%2Cregion%2Ccountry%2Cdivision%2Cage%2Csex%2Chospitalized%2Cdied%2CfullyVaccinated&host=Human&country=Belgium

```
fields: date,region,country,division,age,sex,hospitalized,died,fullyVaccinated
host: Human
country: Belgium
```

If you get rid of them, the requests are fast: https://cov-spectrum.ethz.ch/gisaid/api/v1/sample/aggregated?fields=date%2Cregion%2Ccountry%2Cdivision&host=Human&country=Belgium

It's all that (in my eyes quite unnecessary) baggage of age, sex, hospitalized etc that makes it slow.

Sure, provide them through LAPIS to whoever wants them, and make them show on covSPECTRUM, but only when people really want these graphs.

For 99% of (real) use cases, we don't need: `age, sex, hospitalized, died, fullVaccinated`. That's what makes it all so slow. 

These granularities screw up aggregation. Instead of saving lines by having `count: 50` say, it's all broken up into 50x `count:1`.

We should only request the data we really need, not all the data there is. If I take out division on top it's acceptably fast: https://cov-spectrum.ethz.ch/gisaid/api/v1/sample/aggregated?fields=date%2Cregion%2Ccountry&host=Human

The most important plots can be made without just the simple country request.

If one wants something more detailed, they can wait, but we should not request division data granularly for the whole world. Just for the country that we care about. Same for hospitalization etc, aggregate aggregate aggregate, as much as possible.

If it increases request number by 5, that's fine. Because it reduces downloaded line numbers by a factor of 100.

**Comment by chaoran-chen** [comment]
@corneliusroemer, some requests are certainly (still) slow, but have you observed some extreme cases like 7+ min recently?

**Comment by chaoran-chen** [comment]
The performance issues should have been solved by #380.
